snuba/web/rpc/v1/resolvers/R_eap_spans/common/sampling_in_storage_util.py

```diff
@@ -52,64 +76,96 @@ def _get_most_downsampled_tier() -> Tier:
     return sorted(Tier, reverse=True)[0]
 
 
+def _record_value_in_span_and_DD(
+    span: Span,
+    metrics_backend_func: MetricsBackendType,
+    name: str,
+    value: float | int,
+    tags: Dict[str, str],
+) -> None:
+    metrics_backend_func(name, value, tags, None)
+    span.set_data(name, value)
+
+
 def _get_target_tier(
-    most_downsampled_res: QueryResult, metrics_backend: MetricsBackend, referrer: str
-) -> Tuple[Tier, float]:
+    most_downsampled_res: QueryResult,
+    metrics_backend: MetricsBackend,
+    referrer: str,
+    timer: Timer,
+) -> Tuple[Tier, int]:
+    _ESTIMATION_START_MARK = _SAMPLING_IN_STORAGE_PREFIX + "estimation_start"
+    _ESTIMATION_END_MARK = _SAMPLING_IN_STORAGE_PREFIX + "estimation_end"
```

Comment:
the estimation time is not dominated by this, it's dominated by the time it took to run the query on the most downsampled tier